### ISSUE TITLE
fix: `@IsDateString()` decorator is not validating dates like 2019-02-31

### ISSUE THREAD

**Issue by eleazar1595** [typestack/class-validator#1601]
### Description
The decorator `@IsDateString()` is not validating dates like `2019-02-31`

**Minimal code-snippet showcasing the problem**

## Here the original source code, please see the big arrow indicating where the error is.
```ts
export function isDateString(value: unknown, options?: ValidatorJS.IsISO8601Options): boolean {
  return isISO8601(value, options);
}

/**
 * Alias for IsISO8601 validator
 */
export function IsDateString(
  options?: ValidatorJS.IsISO8601Options,
  validationOptions?: ValidationOptions
): PropertyDecorator {
  return ValidateBy(
    {
      name: IS_DATE_STRING,
      constraints: [options],
      validator: {
        validate: (value, args): boolean => isDateString(value), //<============ HERE THE options argument IS MISSING.
        defaultMessage: buildMessage(
          eachPrefix => eachPrefix + '$property must be a valid ISO 8601 date string',
          validationOptions
        ),
      },
    },
    validationOptions
  );
}
```

### Expected behavior
Invalidates dates like `2019-02-31`.

### Actual behavior
It doesn't invalidate dates like `2019-02-31`.

**Comment by jingyuexing** [comment]
```js
let dateString =  /\d{2,4}(\-\d{1,2}){2}/
```

**Comment by braaar** [comment]
`IsDateString` is an alias for `Is8601`, which matches the [ISO 8601 date format](https://en.wikipedia.org/wiki/ISO_8601). Have you tried `IsDate`?

**Comment by eleazar1595** [comment]
> `IsDateString` is an alias for `Is8601`, which matches the [ISO 8601 date format](https://en.wikipedia.org/wiki/ISO_8601). Have you tried `IsDate`?

I had to change it to `IsISO8601` while someone fixes this.
The fix is very simple, pass the arg `options`
```javascript
validate: (value, args): boolean => isDateString(value), //<============ HERE THE options argument IS MISSING.
```

**Comment by braaar** [comment]
> > `IsDateString` is an alias for `Is8601`, which matches the [ISO 8601 date format](https://en.wikipedia.org/wiki/ISO_8601). Have you tried `IsDate`?
> 
> I had to change it to `IsISO8601` while someone fixes this. The fix is very simple, pass the arg `options`
> 
> ```js
> validate: (value, args): boolean => isDateString(value), //<============ HERE THE options argument IS MISSING.
> ```

Sorry, I misread the behaviour and though it was validating the bad date. I had it backwards.

Why would adding the options parameter change the behaviour for invalid Iso8601 dates? Shouldn't this work as it is without any options?

**Comment by eleazar1595** [comment]
@braaar 

The arg `options` can contain params like `strict` and `strictSeparator `

The following declared function:
```javascript
export function isDateString(value: unknown, options?: ValidatorJS.IsISO8601Options): boolean {
  return isISO8601(value, options);
}
```
Receives the `options` that will be passed to the function `isISO8601(value, options);`

Since that function (`isDateString`) it is called as follow:
```javascript
validator: {
  validate: (value, args): boolean => isDateString(value), //<============ HERE THE options argument IS MISSING.
  defaultMessage: buildMessage(
    eachPrefix => eachPrefix + '$property must be a valid ISO 8601 date string',
    validationOptions
  ),
}
```

We can see that `isDateString` is called with the `value` only, so at the end, the function `isISO8601` won't receive the necessary options like `strick` and `strictSeparator`.

Because of that, the nested validator lib won't validate wrong dates like `2019-02-31`

**Comment by ecrozatti** [comment]
Great contribution, @eleazar1595 ! I also needed to change from `IsDateString` to `IsISO8601` while this issue is not fixed. Thanks!

**Comment by NoNameProvided** [comment]
This was fixed in `0.14.0`. The below snippet now correctly returns a validation error.

```ts
import { IsDateString, validate } from 'class-validator';

class MyPayload {
  @IsDateString({ strict: true })
  date: string;

  constructor(date?: string) {
    this.date = date;
  }
}

// This will fail
validate(new MyPayload('2019-02-31')).then(console.log);
```
